c/datasetjson.c

```diff
@@ -2251,9 +2254,238 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
 #endif /* __ZOWE_OS_ZOS */
 }
 
+/*************************/
+#define FIELD_MAX_LENGTH    1024
+#define REQUEST_MAX_LENGTH  2048
 
-#endif /* not METTLE - the whole module */
+int qsamAllocFile( qsamFileRequest *request, char* message, 
+                   int messageLength){
+#ifdef __ZOWE_OS_ZOS
+  char *baseRequest = "alloc new catalog msg(2) ";
+  char *freeRequest = "free  ";
+  char tempString[20] = {0};
+
+  char field[FIELD_MAX_LENGTH];
+  int status, value;
+  char *ptr;
+
+  /* Must supply daName */
+  if (request->daName == NULL) {
+    return -1;
+  }
+
+  /* Allocate structure in A31 space */
+  EXTF *bpxwdyn=(EXTF *)fetch("BPXWDYN");
```

Comment:
There is no matching ```release()``` for this fetch.

Reply:
Nothing that I have read has mentioned anything about releasing an item that has been fetched.  What is your source for this?

Reply:
The source is in the [manual](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.bpxbd00/xrelease.htm#xrelease). It won't leak resources because the CDE has a use count but it's always good to clean up. You could stash the ```bpxwdyn``` pointer in a static variable and fetch it once. It will get released when the task ends.

Reply:
Thanks for the info.  None of the examples on bpxwdyn has the release function shown.

Reply:
In fact, I'm not sure that it won't leak memory because there is a fetch control block associated with fetch used by LE. The underlying LOAD macro used by fetch is fine but I would still always call DELETE when I use a LOAD.

Reply:
It's a fetch issue not related to BPXWDYN. fetch is is just a callable service.

Reply:
But I would assume that any example showing how to use bpxwdyn would also show the release of the function since it already show the fetch of it.

Reply:
Thanks, fixed